lib/oli_web/live/workspaces/course_author/activity_bank_live.ex

```diff
@@ -13,44 +12,55 @@ defmodule OliWeb.Workspaces.CourseAuthor.ActivityBankLive do
 
     case BankEditor.create_context(project.slug, author) do
       {:ok, context} ->
-        {:ok,
-         assign(socket,
-           active: :bank,
-           context: context,
-           is_admin?: is_admin?,
-           revision_history_link: is_admin?,
-           scripts: Oli.Activities.get_activity_scripts(),
-           resource_slug: project.slug,
-           resource_title: project.title,
-           ctx: ctx
-         )}
+        scripts = Oli.Activities.get_activity_scripts() |> Enum.map(&"/js/#{&1}")
+
+        assign(socket,
+          scripts_loaded: false,
+          scripts: scripts,
+          active: :bank,
+          context: context,
+          is_admin?: is_admin?,
+          revision_history_link: is_admin?,
+          resource_slug: project.slug,
+          resource_title: project.title,
+          ctx: ctx
+        )
+        |> push_event("load_survey_scripts", %{script_sources: scripts})
+        |> then(fn socket -> {:ok, socket} end)
 
       _ ->
         OliWeb.ResourceController.render_not_found(OliWeb.Endpoint, project.slug)
     end
   end
 
-  @impl Phoenix.LiveView
+  @impl true
   def render(assigns) do
     ~H"""
-    <h2 id="header_id" class="pb-2">Activity Bank</h2>
-    <script type="text/javascript" src={Routes.static_path(OliWeb.Endpoint, "/js/activitybank.js")}>
-    </script>
-
-    <%= for script <- @scripts do %>
-      <script type="text/javascript" src={Routes.static_path(OliWeb.Endpoint, "/js/#{script}")}>
-      </script>
-    <% end %>
-
-    <div id="editor">
-      <%= React.component(
-        @ctx,
-        "Components.ActivityBank",
-        Map.merge(@context, %{revisionHistoryLink: @revision_history_link}),
-        id: "activity-bank"
-      ) %>
+    <div id="eventIntercept" phx-hook="LoadSurveyScripts">
+      <h2 id="header_id" class="pb-2">Activity Bank</h2>
+      <%= if connected?(@socket) and assigns[:scripts_loaded] do %>
+        <div id="editor">
+          <%= React.component(
+            @ctx,
+            "Components.ActivityBank",
+            Map.merge(@context, %{revisionHistoryLink: @revision_history_link}),
+            id: "activity-bank"
+          ) %>
+        </div>
+        <%= React.component(@ctx, "Components.ModalDisplay", %{}, id: "modal-display") %>
+      <% else %>
+        <.loader />
+      <% end %>
     </div>
-    <%= React.component(@ctx, "Components.ModalDisplay", %{}, id: "modal-display") %>
     """
   end
+
+  @impl true
+  def handle_event("survey_scripts_loaded", %{"error" => _}, socket) do
+    {:noreply, assign(socket, error: true)}
```

Comment:
What do you think about removing the spinner and informing the user an error has occurred? If not the user will see the spinner forever. Something like:

```suggestion
    {:noreply, socket |> assign(scripts_loaded: true) |> put_flash(:error, "The page had an error while loading")}
```